cpp/serve/grammar/grammar_state_matcher_base.h

```diff
@@ -126,7 +126,7 @@ inline bool GrammarStateMatcherBase::AcceptCodepoint(TCodepoint codepoint, bool
 
   tmp_new_stack_tops_.clear();
   for (auto prev_top : prev_stack_tops) {
-    const auto& cur_rule_position = tree_[prev_top];
+    auto cur_rule_position = tree_[prev_top];
```

Comment:
We need to avoid keeping a constant reference to a vector element. When the vector is reallocated due to new insertion, the reference will become invalid.